crates/contract/src/storage_slot.rs

```diff
@@ -0,0 +1,134 @@
+use alloy_network::TransactionBuilder;
+use alloy_primitives::{Address, Bytes, B256, U256};
+use alloy_provider::Provider;
+use alloy_rpc_types_eth::{
+    state::{AccountOverride, EvmOverrides, StateOverridesBuilder},
+    TransactionRequest,
+};
+use alloy_sol_types::{sol, SolValue};
+
+/// A future type for finding erc20 storage slot.
+#[derive(Debug, Clone)]
+pub struct StorageSlotFinder<P> {
+    provider: P,
```

Comment:
we also need to make this generic over N: Network, so that we can use `N::TransactionRequest::default`, only then we can use this in combination wiht the AnyNetwork